core/chains/evm/chain_set.go

```diff
@@ -22,52 +22,172 @@ import (
+var _ legacyChainSet = &chainSet{}
 
-//go:generate mockery --quiet --name ChainSet --output ./mocks/ --case=underscore
-type ChainSet interface {
+type legacyChainSet interface {
 	services.ServiceCtx
-	chains.Chains
-	chains.Nodes
+	chains.ChainStatuser
+	chains.NodesStatuser
 
 	Get(id *big.Int) (Chain, error)
 
 	Default() (Chain, error)
 	Chains() []Chain
 	ChainCount() int
 
-	Configs() types.Configs
+	Configs() evmtypes.Configs
 
 	SendTx(ctx context.Context, chainID, from, to string, amount *big.Int, balanceCheck bool) error
 }
 
+type EVMChainRelayerExtender interface {
+	relay.RelayerExt
+	Chain() Chain
+	Default() bool
+}
+
+type EVMChainRelayerExtenderSlicer interface {
+	Slice() []EVMChainRelayerExtender
+	Len() int
+}
+
+type ChainRelayerExtenders struct {
+	exts []EVMChainRelayerExtender
+}
+
+var _ EVMChainRelayerExtenderSlicer = &ChainRelayerExtenders{}
+
+func newChainRelayerExtsFromSlice(exts []*ChainRelayerExt) *ChainRelayerExtenders {
+	temp := make([]EVMChainRelayerExtender, len(exts))
+	for i := range exts {
+		temp[i] = exts[i]
+	}
+	return &ChainRelayerExtenders{
+		exts: temp,
+	}
+
+}
+func (c *ChainRelayerExtenders) Slice() []EVMChainRelayerExtender {
+	return c.exts
+}
+
+func (c *ChainRelayerExtenders) Len() int {
+	return len(c.exts)
+}
+
+// implements OneChain
+type ChainRelayerExt struct {
+	chain Chain
+	// TODO remove this altogether. chainset is an implementation detail
+	// that enables use to reuse config logic but the notion of a chainset is
+	// confusing and unneeded in our desired end state where there is 1:1
+	// relayer:chain
+	// TODO add ticket to track
+	cs        *chainSet
+	isDefault bool
+}
+
+var _ EVMChainRelayerExtender = &ChainRelayerExt{}
+
+func (s *ChainRelayerExt) Chain() Chain {
+	return s.chain
+}
+
+func (s *ChainRelayerExt) Default() bool {
+	return s.isDefault
+}
+
+var ErrCorruptEVMChain = errors.New("corrupt evm chain")
+
+func (s *ChainRelayerExt) Start(ctx context.Context) error {
+	if len(s.cs.chains) > 1 {
+		// perhaps better as a panic?
+		err := fmt.Errorf("%w: internal error more than one chain (%d)", ErrCorruptEVMChain, len(s.cs.chains))
+		panic(err)
+	}
+	return s.cs.Start(ctx)
+}
+
+func (s *ChainRelayerExt) Close() (err error) {
+	return s.cs.Close()
+}
+
+func (s *ChainRelayerExt) Name() string {
+	// we set each private chainSet logger to contain the chain id
+	return s.cs.Name()
+}
+
+func (s *ChainRelayerExt) HealthReport() map[string]error {
+	return s.cs.HealthReport()
+}
+
+func (s *ChainRelayerExt) Ready() (err error) {
+	return s.cs.Ready()
+}
+
+var ErrInconsistentChainRelayerExtender = errors.New("inconsistent evm chain relayer extender")
+
+func (s *ChainRelayerExt) ChainStatus(ctx context.Context, id string) (relaytypes.ChainStatus, error) {
+	// we need to implement the interface, but passing id doesn't really make sense because there is only
+	// one chain here. check the id here to provide clear error reporting.
+	if s.chain.ID().String() != id {
```

Comment:
I had imagined these signatures would be updated as part of this change. Is that just too much at once? Are you planning to follow up?

Reply:
yes, i think that's too much to do at once. i have a ticket for that work, which i'll add to this comment